test/decomposer/test_aba_decomposer.py

```diff
@@ -0,0 +1,44 @@
+from typing import Callable
+
+import numpy as np
+import pytest
+
+from opensquirrel.ir import BlochSphereRotation
+from opensquirrel.passes.decomposer import (
+    Decomposer,
+    XYXDecomposer,
+    XZXDecomposer,
+    YXYDecomposer,
+    YZYDecomposer,
+    ZXZDecomposer,
+    ZYZDecomposer,
+)
+from opensquirrel.passes.decomposer.general_decomposer import check_gate_replacement
+
+DECOMPOSER = [XYXDecomposer, XZXDecomposer, YXYDecomposer, YZYDecomposer, ZXZDecomposer, ZYZDecomposer]
+
+
+@pytest.mark.parametrize("decomposer_class", DECOMPOSER)
+def test_specific_bloch_rotation(decomposer_class: Callable[..., Decomposer]) -> None:
+    decomposer = decomposer_class()
+    axis = [-0.53825, -0.65289, -0.53294]
+    angle = 1.97871
+
+    arbitrary_operation = BlochSphereRotation(qubit=0, axis=axis, angle=angle, phase=0)
+    decomposed_arbitrary_operation = decomposer.decompose(arbitrary_operation)
+    check_gate_replacement(arbitrary_operation, decomposed_arbitrary_operation)
+
+
+@pytest.mark.parametrize("decomposer_class", DECOMPOSER)
```

Comment:
If `test_arbitrary` is no longer needed, I wouldn't use the `DECOMPOSER` constant anymore.

Reply:
I will keep test_specific which does need it